elliottlib/find_cve_flaws.py

```diff
@@ -0,0 +1,76 @@
+import requests
+import ssl
+from elliottlib import constants, bzutil
+from requests_kerberos import HTTPKerberosAuth
+
+
+def get_attached_tracker_bugs(bzapi, advisory_id):
+    return bzapi.getbugs([
+        bug['id']
+        for bug in get_all_attached_bugs(advisory_id)
+        if is_tracker_bug(bug)
+    ])
+
+
+def get_all_attached_bugs(advisory_id):
+    return [bug['bug'] for bug in get_advisory(advisory_id)['bugs']['bugs']]
+
+
+def is_tracker_bug(bug):
+    return 'Security' in bug['keywords'] and 'SecurityTracking' in bug['keywords']
+
+
+def get_advisory(advisory_id):
+    res = requests.get(
+        '{}/api/v1/erratum/{}'.format(constants.errata_url, advisory_id),
+        verify=ssl.get_default_verify_paths().openssl_cafile,
+        auth=HTTPKerberosAuth(),
+    )
+    return res.json()
+
+
+def get_corresponding_flaw_bugs(bzapi, tracker_bugs):
+    return bzapi.getbugs(unique(flatten([t.blocks for t in tracker_bugs])))
+
+
+def is_first_fix(bzapi, flaw_bug, tracker_ids_to_be_ignored=[]):
+    other_flaw_trackers = bzapi.getbugs([
+        t for t in flaw_bug.depends_on
```

Comment:
this is going to retrieve _*all*_ trackers, not just openshift ones. we want to ignore whether openstack etc have fixed the same bug.

Reply:
yes, I'm filtering non-OpenShift bugs later, but it would be better if I could not even retrieve them to begin with. let me see if I can add some filters to `getbugs`

Reply:
hm ... found an alternative to filter them out, but using `build_query` + `query` instead:
```
$ python
Python 3.9.0 (default, Oct  6 2020, 00:00:00) 
[GCC 10.2.1 20200826 (Red Hat 10.2.1-3)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import bugzilla
>>> bz = bugzilla.Bugzilla('bugzilla.redhat.com')
>>> bugs = bz.getbugs([1880454, 1881475, 1880456])
>>> for bug in bugs:
...   print(bug.product)
... 
Security Response
OpenShift Container Platform
Security Response
>>> len(bugs)
3
>>> bugs2 = bz.query(bz.build_query(bug_id=[1880454, 1881475, 1880456], product='OpenShift Container Platform'))
>>> len(bugs2)
1
>>> for bug in bugs2:
...   print('{} {}'.format(bug.id, bug.product))
... 
1881475 OpenShift Container Platform
>>>
```

Reply:
done